src/contexts/Cart/cartReducer.js

```diff
@@ -0,0 +1,18 @@
+export const reducerFn = (state, action) => {
+	let _state = null;
+
+	// NOTE: getting action undefined without using if()
+	if(action){
+		switch (action.type){
+			case "ADD_TO_CART":
+				return {...state, cart: state.cart.concat(action.payload)};
+			case "REMOVE_FROM_CART":
+				_state = {...state};
+				_state.cart = state.cart.filter(item => item.id!==action.payload)
+				return _state;

```

Comment:
I needed to make a new _state because without it useEffect was not detecting any changes to the state, and so the changes were not rendering. By creating _state, we get a new state with a new reference all together, so useEffect then saw this change.

Reply:
I need to discuss this in more detail to understand.